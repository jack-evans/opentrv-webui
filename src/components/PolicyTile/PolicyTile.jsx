import React from 'react'
import { Icon, OverflowMenu, OverflowMenuItem } from 'carbon-components-react'

const PolicyTile = ({ id, name, temp, active, handleDelete }) => {
  let activityClassName = active ? 'PolicyTile__footer-active' : 'PolicyTile__footer-idle'
  let activityCircleClassName = active ? 'PolicyTile__footer-active-circle' : 'PolicyTile__footer-idle-circle'
  let activityContent = active ? 'Active' : 'Idle'

  let href = `/policy/${id}`
  return (
    <div className='PolicyTile'>
      <div className='PolicyTile__overflow-container'>
        <OverflowMenu className='PolicyTile__overflow-menu'>
          <OverflowMenuItem className='PolicyTile__overflow-menu-view' itemText='View' onClick={() => { window.location.assign(href) }} />
          <OverflowMenuItem className='PolicyTile__overflow-menu-delete' itemText='Delete' hasDivider isDelete onClick={handleDelete} />
        </OverflowMenu>
      </div>
      <div className='PolicyTile__center-container' style={{cursor: 'pointer'}} onClick={() => { window.location.assign(href) }}>
        <div className='PolicyTile__name-container'>
          <h3>{name}</h3>
        </div>
        <div className='PolicyTile__icon-container'>
          <div className='PolicyTile__icon-circle'>
            <div>
              <Icon
                name='schematics'
                description='Policys icon'
              />
            </div>
          </div>
        </div>
      </div>
      <div className='PolicyTile__footer'>
        <div className={activityClassName}>
          <div className={activityCircleClassName} />
          <h3>{activityContent}</h3>
        </div>
        <div className='PolicyTile__footer-temp'>
          <h3>{temp}&#176;C</h3>
        </div>
      </div>
    </div>
  )
}

export default PolicyTile
