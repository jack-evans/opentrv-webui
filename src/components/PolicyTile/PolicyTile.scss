
.PolicyTile {
  background-color: $ui-02;
  border: 1px solid $ui-04;
  box-shadow: 0 1px 3px 0 $ui-04;
  height: 250px;
  margin: 40px;
  padding: 10px;
  width: 280px;

  .PolicyTile__overflow-container {
    border-bottom: 1px solid $ui-04;
    display: flex;
    justify-content: flex-end;

    .PolicyTile__overflow-menu {
      outline: none;
    }
  }

  .PolicyTile__name-container {
    display: flex;
    justify-content: flex-start;
    padding-top: 5px;
  }

  .PolicyTile__icon-container {
    align-items: center;
    display: flex;
    height: 120px;
    justify-content: center;

    .PolicyTile__icon-circle {
      align-self: center;
      border: 1px solid $ui-04;
      border-radius: 50%;
      display: flex;
      height: 60px;
      justify-content: center;
      width: 60px;

      div {
        align-self: center;
      }
    }
  }

  .PolicyTile__footer {
    border-top: 1px solid $ui-04;
    display: flex;
    justify-content: space-between;
    padding-top: 10px;

    .PolicyTile__footer-active {
      display: flex;

      .PolicyTile__footer-active-circle {
        align-self: center;
        background-color: $support-02;
        border-radius: 50%;
        height: 20px;
        width: 20px;
      }

      h3 {
        padding-left: 5px;
      }
    }

    .PolicyTile__footer-idle {
      display: flex;

      .PolicyTile__footer-idle-circle {
        align-self: center;
        background-color: $support-01;
        border-radius: 50%;
        height: 20px;
        width: 20px;
      }

      h3 {
        padding-left: 5px;
      }
    }

    .PolicyTile__footer-temp {
      justify-content: flex-end;
    }
  }
}

.PolicyTile:hover {
  box-shadow: 0 4px 8px 0 $ui-05;
}