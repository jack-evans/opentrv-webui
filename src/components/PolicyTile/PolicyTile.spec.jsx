import React from 'react'
import PolicyTile from './PolicyTile'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('PolicyTile.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<PolicyTile id='1234' name='test' temp={26.3} active={false} />)).toMatchSnapshot()
  })

  describe('when the Policy is not active (false)', () => {
    it('sets the footer class to PolicyTile__footer-idle', () => {
      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active={false} />)
      expect(wrapper.find('.PolicyTile__footer-idle')).toHaveLength(1)
    })

    it('sets the activity circle class to PolicyTile__footer-idle-circle', () => {
      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active={false} />)
      expect(wrapper.find('.PolicyTile__footer-idle-circle')).toHaveLength(1)
    })

    it('sets the activity content to be "Idle"', () => {
      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active={false} />)
      expect(wrapper.text()).toContain('Idle')
    })
  })

  describe('when the Policy is active (true)', () => {
    it('sets the footer class to PolicyTile__footer-active', () => {
      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
      expect(wrapper.find('.PolicyTile__footer-active')).toHaveLength(1)
    })

    it('sets the activity circle class to PolicyTile__footer-active-circle', () => {
      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
      expect(wrapper.find('.PolicyTile__footer-active-circle')).toHaveLength(1)
    })

    it('sets the activity content to be "Active"', () => {
      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
      expect(wrapper.text()).toContain('Active')
    })
  })

  describe('when the PolicyTile is clicked on', () => {
    it('calls the assign method for window.location', () => {
      global.location.assign = jest.fn()

      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
      wrapper.find('.PolicyTile__center-container').simulate('click')

      expect(global.location.assign).toHaveBeenCalledTimes(1)
    })

    it('calls the assign method for window.location', () => {
      global.location.assign = jest.fn()

      const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
      wrapper.find('.PolicyTile__center-container').simulate('click')

      expect(global.location.assign).toHaveBeenCalledWith('/policys/1234')
    })
  })

  describe('when the PolicyTile overflow menu is clicked on', () => {
    describe('when view is clicked on', () => {
      it('calls the assign method for window.location', () => {
        global.location.assign = jest.fn()

        const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
        wrapper.find('.PolicyTile__overflow-menu-view').simulate('click')

        expect(global.location.assign).toHaveBeenCalledTimes(1)
      })

      it('calls the assign method for window.location', () => {
        global.location.assign = jest.fn()

        const wrapper = shallow(<PolicyTile id='1234' name='test' temp={26.3} active />)
        wrapper.find('.PolicyTile__overflow-menu-view').simulate('click')

        expect(global.location.assign).toHaveBeenCalledWith('/policys/1234')
      })
    })
  })
})
