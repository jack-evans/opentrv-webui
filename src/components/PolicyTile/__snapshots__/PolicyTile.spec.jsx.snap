// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyTile.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PolicyTile
    active={false}
    id="1234"
    name="test"
    temp={26.3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="PolicyTile__overflow-container"
        >
          <OverflowMenu
            ariaLabel="list of options"
            className="PolicyTile__overflow-menu"
            flipped={false}
            floatingMenu={false}
            iconDescription="open and close list of options"
            iconName="overflow-menu"
            menuOffset={[Function]}
            menuOffsetFlip={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            open={false}
            tabIndex={0}
          >
            <OverflowMenuItem
              className="PolicyTile__overflow-menu-view"
              disabled={false}
              hasDivider={false}
              isDelete={false}
              itemText="View"
              onClick={[Function]}
            />
            <OverflowMenuItem
              className="PolicyTile__overflow-menu-delete"
              disabled={false}
              hasDivider={true}
              isDelete={true}
              itemText="Delete"
              onClick={[Function]}
            />
          </OverflowMenu>
        </div>,
        <div
          className="PolicyTile__center-container"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            className="PolicyTile__name-container"
          >
            <h3>
              test
            </h3>
          </div>
          <div
            className="PolicyTile__icon-container"
          >
            <div
              className="PolicyTile__icon-circle"
            >
              <div>
                <Icon
                  description="Policys icon"
                  fillRule="evenodd"
                  name="schematics"
                  role="img"
                />
              </div>
            </div>
          </div>
        </div>,
        <div
          className="PolicyTile__footer"
        >
          <div
            className="PolicyTile__footer-idle"
          >
            <div
              className="PolicyTile__footer-idle-circle"
            />
            <h3>
              Idle
            </h3>
          </div>
          <div
            className="PolicyTile__footer-temp"
          >
            <h3>
              26.3
              Â°C
            </h3>
          </div>
        </div>,
      ],
      "className": "PolicyTile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <OverflowMenu
            ariaLabel="list of options"
            className="PolicyTile__overflow-menu"
            flipped={false}
            floatingMenu={false}
            iconDescription="open and close list of options"
            iconName="overflow-menu"
            menuOffset={[Function]}
            menuOffsetFlip={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            open={false}
            tabIndex={0}
          >
            <OverflowMenuItem
              className="PolicyTile__overflow-menu-view"
              disabled={false}
              hasDivider={false}
              isDelete={false}
              itemText="View"
              onClick={[Function]}
            />
            <OverflowMenuItem
              className="PolicyTile__overflow-menu-delete"
              disabled={false}
              hasDivider={true}
              isDelete={true}
              itemText="Delete"
              onClick={[Function]}
            />
          </OverflowMenu>,
          "className": "PolicyTile__overflow-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ariaLabel": "list of options",
            "children": Array [
              <OverflowMenuItem
                className="PolicyTile__overflow-menu-view"
                disabled={false}
                hasDivider={false}
                isDelete={false}
                itemText="View"
                onClick={[Function]}
              />,
              <OverflowMenuItem
                className="PolicyTile__overflow-menu-delete"
                disabled={false}
                hasDivider={true}
                isDelete={true}
                itemText="Delete"
                onClick={[Function]}
              />,
            ],
            "className": "PolicyTile__overflow-menu",
            "flipped": false,
            "floatingMenu": false,
            "iconDescription": "open and close list of options",
            "iconName": "overflow-menu",
            "menuOffset": [Function],
            "menuOffsetFlip": [Function],
            "onClick": [Function],
            "onKeyDown": [Function],
            "open": false,
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "PolicyTile__overflow-menu-view",
                "disabled": false,
                "hasDivider": false,
                "isDelete": false,
                "itemText": "View",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "PolicyTile__overflow-menu-delete",
                "disabled": false,
                "hasDivider": true,
                "isDelete": true,
                "itemText": "Delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="PolicyTile__name-container"
            >
              <h3>
                test
              </h3>
            </div>,
            <div
              className="PolicyTile__icon-container"
            >
              <div
                className="PolicyTile__icon-circle"
              >
                <div>
                  <Icon
                    description="Policys icon"
                    fillRule="evenodd"
                    name="schematics"
                    role="img"
                  />
                </div>
              </div>
            </div>,
          ],
          "className": "PolicyTile__center-container",
          "onClick": [Function],
          "style": Object {
            "cursor": "pointer",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                test
              </h3>,
              "className": "PolicyTile__name-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="PolicyTile__icon-circle"
              >
                <div>
                  <Icon
                    description="Policys icon"
                    fillRule="evenodd"
                    name="schematics"
                    role="img"
                  />
                </div>
              </div>,
              "className": "PolicyTile__icon-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  <Icon
                    description="Policys icon"
                    fillRule="evenodd"
                    name="schematics"
                    role="img"
                  />
                </div>,
                "className": "PolicyTile__icon-circle",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Icon
                    description="Policys icon"
                    fillRule="evenodd"
                    name="schematics"
                    role="img"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "description": "Policys icon",
                    "fillRule": "evenodd",
                    "name": "schematics",
                    "role": "img",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="PolicyTile__footer-idle"
            >
              <div
                className="PolicyTile__footer-idle-circle"
              />
              <h3>
                Idle
              </h3>
            </div>,
            <div
              className="PolicyTile__footer-temp"
            >
              <h3>
                26.3
                Â°C
              </h3>
            </div>,
          ],
          "className": "PolicyTile__footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="PolicyTile__footer-idle-circle"
                />,
                <h3>
                  Idle
                </h3>,
              ],
              "className": "PolicyTile__footer-idle",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "PolicyTile__footer-idle-circle",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Idle",
                },
                "ref": null,
                "rendered": "Idle",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                26.3
                Â°C
              </h3>,
              "className": "PolicyTile__footer-temp",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  26.3,
                  "Â°C",
                ],
              },
              "ref": null,
              "rendered": Array [
                26.3,
                "Â°C",
              ],
              "type": "h3",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="PolicyTile__overflow-container"
          >
            <OverflowMenu
              ariaLabel="list of options"
              className="PolicyTile__overflow-menu"
              flipped={false}
              floatingMenu={false}
              iconDescription="open and close list of options"
              iconName="overflow-menu"
              menuOffset={[Function]}
              menuOffsetFlip={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              open={false}
              tabIndex={0}
            >
              <OverflowMenuItem
                className="PolicyTile__overflow-menu-view"
                disabled={false}
                hasDivider={false}
                isDelete={false}
                itemText="View"
                onClick={[Function]}
              />
              <OverflowMenuItem
                className="PolicyTile__overflow-menu-delete"
                disabled={false}
                hasDivider={true}
                isDelete={true}
                itemText="Delete"
                onClick={[Function]}
              />
            </OverflowMenu>
          </div>,
          <div
            className="PolicyTile__center-container"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="PolicyTile__name-container"
            >
              <h3>
                test
              </h3>
            </div>
            <div
              className="PolicyTile__icon-container"
            >
              <div
                className="PolicyTile__icon-circle"
              >
                <div>
                  <Icon
                    description="Policys icon"
                    fillRule="evenodd"
                    name="schematics"
                    role="img"
                  />
                </div>
              </div>
            </div>
          </div>,
          <div
            className="PolicyTile__footer"
          >
            <div
              className="PolicyTile__footer-idle"
            >
              <div
                className="PolicyTile__footer-idle-circle"
              />
              <h3>
                Idle
              </h3>
            </div>
            <div
              className="PolicyTile__footer-temp"
            >
              <h3>
                26.3
                Â°C
              </h3>
            </div>
          </div>,
        ],
        "className": "PolicyTile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <OverflowMenu
              ariaLabel="list of options"
              className="PolicyTile__overflow-menu"
              flipped={false}
              floatingMenu={false}
              iconDescription="open and close list of options"
              iconName="overflow-menu"
              menuOffset={[Function]}
              menuOffsetFlip={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              open={false}
              tabIndex={0}
            >
              <OverflowMenuItem
                className="PolicyTile__overflow-menu-view"
                disabled={false}
                hasDivider={false}
                isDelete={false}
                itemText="View"
                onClick={[Function]}
              />
              <OverflowMenuItem
                className="PolicyTile__overflow-menu-delete"
                disabled={false}
                hasDivider={true}
                isDelete={true}
                itemText="Delete"
                onClick={[Function]}
              />
            </OverflowMenu>,
            "className": "PolicyTile__overflow-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "ariaLabel": "list of options",
              "children": Array [
                <OverflowMenuItem
                  className="PolicyTile__overflow-menu-view"
                  disabled={false}
                  hasDivider={false}
                  isDelete={false}
                  itemText="View"
                  onClick={[Function]}
                />,
                <OverflowMenuItem
                  className="PolicyTile__overflow-menu-delete"
                  disabled={false}
                  hasDivider={true}
                  isDelete={true}
                  itemText="Delete"
                  onClick={[Function]}
                />,
              ],
              "className": "PolicyTile__overflow-menu",
              "flipped": false,
              "floatingMenu": false,
              "iconDescription": "open and close list of options",
              "iconName": "overflow-menu",
              "menuOffset": [Function],
              "menuOffsetFlip": [Function],
              "onClick": [Function],
              "onKeyDown": [Function],
              "open": false,
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "PolicyTile__overflow-menu-view",
                  "disabled": false,
                  "hasDivider": false,
                  "isDelete": false,
                  "itemText": "View",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "PolicyTile__overflow-menu-delete",
                  "disabled": false,
                  "hasDivider": true,
                  "isDelete": true,
                  "itemText": "Delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="PolicyTile__name-container"
              >
                <h3>
                  test
                </h3>
              </div>,
              <div
                className="PolicyTile__icon-container"
              >
                <div
                  className="PolicyTile__icon-circle"
                >
                  <div>
                    <Icon
                      description="Policys icon"
                      fillRule="evenodd"
                      name="schematics"
                      role="img"
                    />
                  </div>
                </div>
              </div>,
            ],
            "className": "PolicyTile__center-container",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  test
                </h3>,
                "className": "PolicyTile__name-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                },
                "ref": null,
                "rendered": "test",
                "type": "h3",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="PolicyTile__icon-circle"
                >
                  <div>
                    <Icon
                      description="Policys icon"
                      fillRule="evenodd"
                      name="schematics"
                      role="img"
                    />
                  </div>
                </div>,
                "className": "PolicyTile__icon-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div>
                    <Icon
                      description="Policys icon"
                      fillRule="evenodd"
                      name="schematics"
                      role="img"
                    />
                  </div>,
                  "className": "PolicyTile__icon-circle",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Icon
                      description="Policys icon"
                      fillRule="evenodd"
                      name="schematics"
                      role="img"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "description": "Policys icon",
                      "fillRule": "evenodd",
                      "name": "schematics",
                      "role": "img",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="PolicyTile__footer-idle"
              >
                <div
                  className="PolicyTile__footer-idle-circle"
                />
                <h3>
                  Idle
                </h3>
              </div>,
              <div
                className="PolicyTile__footer-temp"
              >
                <h3>
                  26.3
                  Â°C
                </h3>
              </div>,
            ],
            "className": "PolicyTile__footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="PolicyTile__footer-idle-circle"
                  />,
                  <h3>
                    Idle
                  </h3>,
                ],
                "className": "PolicyTile__footer-idle",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "PolicyTile__footer-idle-circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Idle",
                  },
                  "ref": null,
                  "rendered": "Idle",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  26.3
                  Â°C
                </h3>,
                "className": "PolicyTile__footer-temp",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    26.3,
                    "Â°C",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  26.3,
                  "Â°C",
                ],
                "type": "h3",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
